File: controller/app/modify_pwd/form/ModifyPwdAppForm.kt
package form

import io.swagger.v3.oas.annotations.Parameter
import jakarta.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length

type ModifyPwdAppForm struct{

    @Parameter(description = "旧密码")
    @NotBlank
    @Length(min = 4, max = 32)
    oldPwd string

    @Parameter(description = "新密码")
    @NotBlank
    @Length(min = 4, max = 32)
    pwd string
}

File: controller/app/modify_pwd/form/ModifyPwdAppForm.go
package form

import (
	"DairoDFS/dao/UserDao"
	"DairoDFS/extension/String"
	"DairoDFS/util/LoginState"
)

type ModifyPwdAppForm struct {

	/** 旧密码 **/
	//@NotBlank
	//@Length(min = 4, max = 32)
	OldPwd string

	/** 新密码 **/
	//@NotBlank
	//@Length(min = 4, max = 32)
	Pwd string
}

// 验证旧密码
func (mine ModifyPwdAppForm) IsOldPwd() string {
	userId := LoginState.LoginId()
	user, _ := UserDao.SelectOne(userId)
	if user.Pwd != String.ToMd5(mine.OldPwd) {
		return "旧密码不正确"
	}
	return ""
}
